src/Compilers/CSharp/Test/Semantic/Semantics/RecordTests.cs

```diff
@@ -4117,6 +4120,776 @@ End Class
+  IL_0016:  ret
+}");
+            verifier.VerifyIL("C..ctor(C)",
+@"{
+  // Code size       32 (0x20)
+  .maxstack  2
+  IL_0000:  ldarg.0
+  IL_0001:  ldarg.1
+  IL_0002:  call       ""B..ctor(B)""
+  IL_0007:  ldarg.0
+  IL_0008:  ldarg.1
+  IL_0009:  ldfld      ""object C.<X>k__BackingField""
+  IL_000e:  stfld      ""object C.<X>k__BackingField""
+  IL_0013:  ldarg.0
+  IL_0014:  ldarg.1
+  IL_0015:  ldfld      ""object C.<Y>k__BackingField""
+  IL_001a:  stfld      ""object C.<Y>k__BackingField""
+  IL_001f:  ret
+}");
+            verifier.VerifyIL("C.Deconstruct",
+@"{
+  // Code size       17 (0x11)
+  .maxstack  2
+  IL_0000:  ldarg.1
+  IL_0001:  ldarg.0
+  IL_0002:  callvirt   ""object A.X.get""
+  IL_0007:  stind.ref
+  IL_0008:  ldarg.2
+  IL_0009:  ldarg.0
+  IL_000a:  callvirt   ""object A.Y.get""
+  IL_000f:  stind.ref
+  IL_0010:  ret
+}");
+            verifier.VerifyIL("C.Equals(C)",
+@"{
+  // Code size       58 (0x3a)
+  .maxstack  3
+  IL_0000:  ldarg.0
+  IL_0001:  ldarg.1
+  IL_0002:  call       ""bool B.Equals(B)""
+  IL_0007:  brfalse.s  IL_0038
+  IL_0009:  call       ""System.Collections.Generic.EqualityComparer<object> System.Collections.Generic.EqualityComparer<object>.Default.get""
+  IL_000e:  ldarg.0
+  IL_000f:  ldfld      ""object C.<X>k__BackingField""
+  IL_0014:  ldarg.1
+  IL_0015:  ldfld      ""object C.<X>k__BackingField""
+  IL_001a:  callvirt   ""bool System.Collections.Generic.EqualityComparer<object>.Equals(object, object)""
+  IL_001f:  brfalse.s  IL_0038
+  IL_0021:  call       ""System.Collections.Generic.EqualityComparer<object> System.Collections.Generic.EqualityComparer<object>.Default.get""
+  IL_0026:  ldarg.0
+  IL_0027:  ldfld      ""object C.<Y>k__BackingField""
+  IL_002c:  ldarg.1
+  IL_002d:  ldfld      ""object C.<Y>k__BackingField""
+  IL_0032:  callvirt   ""bool System.Collections.Generic.EqualityComparer<object>.Equals(object, object)""
+  IL_0037:  ret
+  IL_0038:  ldc.i4.0
+  IL_0039:  ret
+}");
+            verifier.VerifyIL("C.GetHashCode()",
+@"{
+  // Code size       53 (0x35)
+  .maxstack  3
+  IL_0000:  ldarg.0
+  IL_0001:  call       ""int B.GetHashCode()""
+  IL_0006:  ldc.i4     0xa5555529
+  IL_000b:  mul
+  IL_000c:  call       ""System.Collections.Generic.EqualityComparer<object> System.Collections.Generic.EqualityComparer<object>.Default.get""
+  IL_0011:  ldarg.0
+  IL_0012:  ldfld      ""object C.<X>k__BackingField""
+  IL_0017:  callvirt   ""int System.Collections.Generic.EqualityComparer<object>.GetHashCode(object)""
+  IL_001c:  add
+  IL_001d:  ldc.i4     0xa5555529
+  IL_0022:  mul
+  IL_0023:  call       ""System.Collections.Generic.EqualityComparer<object> System.Collections.Generic.EqualityComparer<object>.Default.get""
+  IL_0028:  ldarg.0
+  IL_0029:  ldfld      ""object C.<Y>k__BackingField""
+  IL_002e:  callvirt   ""int System.Collections.Generic.EqualityComparer<object>.GetHashCode(object)""
+  IL_0033:  add
+  IL_0034:  ret
+}");
+        }
```

Comment:
>} [](start = 8, length = 1)

It might worth testing scenario when an abstract property gets shadowed in the intermediate base by some other member with the same name.

Reply:
Added a test but avoided overriding because the intermediate base members result in errors.

---
In reply to: [443766696](https://github.com/dotnet/roslyn/pull/45336#discussion_r443766696) [](ancestors = 443766696)